### ISSUE TITLE
OS.get_unix_time() works different in _ready/_init than in _process

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#44303]
**Godot version:**
3.2.4.beta.custom_build. efc373bc3

**OS/device including version:**
Ubuntu 20.04
Nvidia GTX 970 - 440

**Issue description:**
Sometimes this simple code crashes in assert line.
The strangest part is, that current_run_time variable is usually smaller than `-50`
 
```
extends Node

var start_time : float
var last_time : float

func _ready(): 
	start_time = OS.get_unix_time()

func _process(delta: float) -> void:
	var current_run_time : float = OS.get_unix_time() - start_time
	assert(current_run_time > 0.0)
	
	print(OS.get_unix_time())
	print(start_time)
	print(current_run_time)
```

current_run_time should have start always with 0 when running project and should have increased value every frame, but don't know why almost always it have different value

This bug doesn't exists in master branch

**Steps to reproduce:**
1. Run minimal project several times until assert will work
(In gif is cut, because I wait a while)
![jfile](https://user-images.githubusercontent.com/41945903/101931226-6c8c2d00-3bd9-11eb-951a-8ff2972a8ebc.gif)

**Minimal reproduction project:**
[32.zip](https://github.com/godotengine/godot/files/5680469/32.zip)

**Comment by Calinou** [comment]
> current_run_time should have start always with 0 when running project and should have increased value every frame, but don't know why almost always it have different value

Maybe some variable isn't initialized in the C++ code?

**Comment by akien-mga** [comment]
It seems to mean that you can have a `_process` run before `_ready`.

**Comment by qarmin** [comment]
I created simpler code which only prints values which shows that `_ready` is executed before `_process`, but still it prints in that between running both functions was ~13 seconds difference which is not true
```
extends Node

var start_time : float

func _ready(): 
	start_time = OS.get_unix_time()
	print("I'm in _ready")

func _process(delta: float) -> void:
	print("I'm in _process")
	var current_run_time : float = OS.get_unix_time() - start_time
	print(OS.get_unix_time())
	print(start_time)
	print(current_run_time)
	get_tree().quit()

```
![Zrzut ekranu z 2020-12-11 18-04-16](https://user-images.githubusercontent.com/41945903/101932704-7b73df00-3bdb-11eb-9132-2dcd462a2787.png)

Minimal project [32.zip](https://github.com/godotengine/godot/files/5680546/32.zip)

**Comment by Demindiro** [comment]
I think the issue is that OS.get_unix_time() returns an int, but instead of converting it to a 64bit float (with 53 bits of precision, well in range of the current epoch time) it is converted to a 32bit float first (24bit precision).

The reason I suspect this is because current_run_time_i counts up by 1 every second and it happens to wrap around at 65 to -63 (... 64, 65, -63, -62, ...), which is a 6-bit number. It also so happens that the current epoch is smaller than 2^31 but larger than 2^30 and 30 - 24 == 6.

**Comment by Faless** [comment]
I understand the issue is about something specific but please:

**Do not use `OS.get_unix_time()` for those things**, since it's subject to NTP adjustments.
**Always use `OS.get_ticks_msec()` or `OS.get_ticks_msec()`.**

**Two consecutive calls to `OS.get_unix_time()` might even end up having the second value lower then the first**

**Comment by akien-mga** [comment]
This would be worth calling out in the documentation for `OS.get_unix_time()` I think.

**Comment by Faless** [comment]
> This would be worth calling out in the documentation for `OS.get_unix_time()` I think.

I'm on it :)

**Comment by qarmin** [comment]
I used it before, but there was removed in 4.0.

**Comment by Faless** [comment]
> Actually I used `get_system_time_msecs`.

Which also returns the wall clock time, not a monotonic value.

**Comment by akien-mga** [comment]
Fixed by #44306.
